manifests/unit_file.pp

```diff
@@ -9,6 +9,25 @@
 ) {
   include ::systemd
 
+  if $title =~ /^([^\/]+)\// {
+    $subdir = $1
+    if $subdir !~ /\.d$/ {
+      fail('drop-in unit dirs must have suffix .d')
+    }
+    if $title !~ /\.conf$/ {
+      fail('drop-in unit files must have suffix .conf')
+    }
+    file { "${path}/${subdir}":
```

Comment:
You are right. We don't do that, so I hadn't noticed. We do override multiple configuration options of a single service, but we put those in a single drop-in unit file. I wonder if there is an actual use case for multiple drop-in unit files for the same service but that is a different discussion.
That situation could be solved by replacing the file{} with an exec{} that does "mkdir -p". Would you like me to do that or would you rather reject the whole patch?